tests/unit/executor/test_step_handler.py

```diff
@@ -95,9 +102,51 @@ def test_run_builtin_build(self, new_dir, mocker):
         Path("parts/p1/run").mkdir(parents=True)
         sh = _step_handler_for_step(Step.BUILD, cache_dir=new_dir)
         result = sh.run_builtin()
+        build_script_path = Path(new_dir / "parts/p1/run/build.sh")
+        environment_script_path = Path(new_dir / "parts/p1/run/environment.sh")
+
+        assert get_mode(environment_script_path) == 0o644
+        with open(environment_script_path, "r") as file:
+            assert file.read() == dedent(
+                f"""\
+                #!/bin/bash
+                set -euo pipefail
+                # Environment
```

Comment:
Now that this resides in a file of its own that's sourced from a different script we could remove these lines and leave only variable exporting, WDYT?

Reply:
Oops, I didn't post my comment from yesterday.
Yes, I agreer.  I don't think there's any need for the first 2 lines anymore.

This appears to work for both the builtin `build` and for build scriplets.